pkg/clients/aws.go

```diff
@@ -632,3 +631,27 @@ func DiffLabels(local, remote map[string]string) (addOrModify map[string]string,
 	}
 	return
 }
+
+type cleanedError struct {
+	code    string
+	message string
+}
+
+func (e cleanedError) Error() string {
+	return fmt.Sprintf("%s - %s", e.code, e.message)
+}
+
+func cleanError(err error) error {
+	if awsErr, ok := err.(awserr.Error); ok {
+		return cleanedError{awsErr.Code(), awsErr.Message()}
```

Comment:
What I feel kind of uneasy about this solution is that we can't use any `Is*Error` check functions we have for each resource since it doesn't satisfy `awserr.Error` interface anymore. The usage of this `awsclient.Wrap` seems to be mostly when returned from the controller anyway, but I feel that we'll see more use of it across the codebase.

Is there a way to keep the returned object from `Wrap` satisfy `awserr.Error` and close to the original error so that we don't need to think about this additional layer of abstraction? Maybe `awserr.New`?

Reply:
Did some testing and was able to get `cleanError` to return a `awserr.Error` but the underlying `errors.Wrap` makes it so it won't satisfy `awserr.Error` anymore.

Thoughts on if I just allow cleanError to be exported and it will return a `awserr.Error` so that is available if you just want to strip the requestId? But still keep the wrap function for ease of use?

Reply:
Yeah I think that'd be nice.
>But still keep the wrap function for ease of use?

I'm not sure. Assuming I got what you're saying right, the usage would be something like this `errors.Wrap(awsclient.CleanError(err), "some error")`, right?

Could you share a code snippet of how it'd look like just to be sure?

Reply:
Oh sorry, was just trying to say wasn't sure If I should keep the `awsclient.Wrap` I had added or remove it and go through and remove  them all and just change to the now exported `CleanError`? 


So itd still be
```
	if err != nil {
		return managed.ExternalObservation{}, awsclient.Wrap(resource.Ignore(iam.IsErrorNotFound, err), errGet)
	}
```
vs
```
if err != nil {
		return managed.ExternalObservation{}, errrors.Wrap(awsclient.CleanError(resource.Ignore(iam.IsErrorNotFound, err)), errGet)
	}
```

Just wasn't sure which way to go and wanted to double check before I go through all of them again

Reply:
No worries. I feel like the second one is easier to grasp for a newcomer to see what we're doing but I don't feel strongly so I'm fine with what you choose here.